apimodels/cedar_models.go

```diff
@@ -1,9 +1,155 @@
 package apimodels
 
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model/log"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/evergreen-ci/timber"
+	"github.com/evergreen-ci/timber/buildlogger"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/pkg/errors"
+)
+
 type CedarConfig struct {
 	BaseURL  string `json:"base_url"`
 	RPCPort  string `json:"rpc_port"`
 	Username string `json:"username"`
 	APIKey   string `json:"api_key,omitempty"`
 	Insecure bool   `json:"insecure"`
 }
+
+// GetBuildloggerLogsOptions represents the arguments passed into the
+// GetBuildloggerLogs function.
+// TODO (EVG-20019): Remove the "V2" suffix once log fetching is consolidated
+// and the original GetBuildloggerLogsOptions struct is deleted.
+type GetBuildloggerLogsOptionsV2 struct {
+	BaseURL   string   `json:"_"`
+	TaskID    string   `json:"-"`
+	Execution *int     `json:"-"`
+	TestName  string   `json:"-"`
+	Tags      []string `json:"-"`
+	Start     int64    `json:"-"`
+	End       int64    `json:"-"`
+	Limit     int      `json:"-"`
+	Tail      int      `json:"-"`
+}
+
+// GetBuildloggerLogs makes request to Cedar for a specifc log and returns a
+// log iterator.
+// TODO (EVG-20019): Remove the "V2" suffix once log fetching is consolidated
+// and the original GetBuildloggerLogs function is deleted.
+func GetBuildloggerLogsV2(ctx context.Context, opts GetBuildloggerLogsOptionsV2) (log.LogIterator, error) {
```

Comment:
Is there a way to test this?

Reply:
Yeah I can write tests for this! We never wrote tests for [the code](https://github.com/evergreen-ci/evergreen/blob/main/apimodels/task_log.go#L126-L184) that this replaces or for any code that lives in `apimodels` for that matter, I just always test in staging. Since it is only reading data for rendering, the fixes are quick to catch in staging and easy to fix if a bug does go out.

Reply:
SGTM given that this code will be deleted in a year.